### PR TITLE
docs(plan): analyze and stage ADR-0038 recovery placement implementation

### PR DESCRIPTION
**Pull request by Szczepanov**
## Stack

**Stacked on #455** (`docs/adr-0038-weekly-recovery-placement`). This PR is intentionally documentation-only; it does not activate ADR-0038 while the ADR remains `Proposed`.

## Summary

Adds and reviews the code-level implementation analysis plus execution plan for ADR-0038 recovery placement:

- `docs/analysis/2026-09-07-adr-0038-recovery-placement-implementation-analysis.md`
- `docs/plans/adr-0038-recovery-placement.md`

The original analysis was traced against the earlier #455 head `45f5652e96571874d3be900cf3aa42455b902e41`; the mutable execution plan has since been aligned with the reviewed parent ADR and the branch now contains current #455 head `957f0265499d2082536df64afc0682ad460ea7e7`.

## Key findings / contracts

1. **Plan-less coverage needs recovery-specific identity authority.** The current empty `CoverageState` cannot classify exact recovery candidates without phase/descriptor authority. Evergreen/general should be used only as product-policy recovery identity authority, never as a fabricated authored plan/block.
2. **A rolling counter is not a placement deadline.** Latest qualifying recovery `R` gives `dueByDate = R + 7` and the due date is tier 1 before candidate selection.
3. **Bootstrap is explicit in both #455 and this plan.** With unknown history, durable bootstrap `B` is the initial deadline reference, `dueByDate = B + 7`; `B` is not recovery credit, the pre-bootstrap unknown prefix is excluded, and the first enforceable window is `B + 1 ... B + 7`.
4. **Performed recovery must retain replayable exact identity.** Canonical `workoutId` plus the qualification authority is required; reverse-derived `templateId` is optional metadata because the existing performed-facts path intentionally treats some workout→template mappings as ambiguous.
5. **Historical complete Rest needs positive day-outcome truth.** Generated Rest requires persisted recommendation/audit plus authoritative post-day occurrence reconciliation; missing telemetry alone never creates Rest.
6. **ADR-0035 provenance has one canonical persisted contract.** `ExternalRestProvenance` remains the audit/replay identity; `ExternalRestDecisionProvenance` is only its override-aware decision extension.
7. **D-SUPPORT has a single-candidate bypass.** Forecast viability currently depends on `ranked.length > 1`; ADR-0038 must remove that shortcut for applicable train/modify discretionary picks.
8. **Simulation must prove placement by exact identity.** Category-based `restOrRecoveryDayCount` is only a smoke metric; the implementation needs exact rolling-window evaluation, max streak, bootstrap-prefix semantics and typed miss/defer diagnostics.
9. **The one-in-seven scalar remains product policy, not physiology.** Knowledge governance should register the heuristic while retaining `recovery.training.stress_recovery_balance` as the scientific boundary.

## Implementation structure

Dependency-ordered packages:

- **RP0** — pure recovery-placement state, deadline/bootstrap authority and typed diagnostics
- **RP1** — exact recovery identity and descriptor consistency
- **RP2** — replayable performed-recovery facts, ADR-0035 authored-rest bridge, generated-Rest day outcome and durable bootstrap persistence
- **RP3** — staged ranking tier integration + single-candidate D-SUPPORT correction; production behavior remains off
- **RP4** — typed observability + deterministic exact-identity rolling seven-date simulation/corpus proof
- **RP5A** — pre-activation Sports Knowledge Registry / inventory / alignment tests
- **RP5B** — final live activation + `POLICY_VERSION`/replay update only after RP4 proof

Recommended runtime stack remains four reviewable PRs:

- **A:** policy state + exact identity
- **B:** historical recovery truth
- **C:** staged ranking + D-SUPPORT + knowledge alignment (**no live switch, no policy-version bump**)
- **D:** deterministic simulation/corpus proof, then live composition switch + `POLICY_VERSION` bump in the same final review boundary

## Remaining decision gate before activation

If ADR-0035 authored Rest exists, no explicit app override was recorded, but canonical performed training later appears on that date, ADR-0038 must explicitly decide whether authored-rest recovery credit survives. The plan recommends treating contradictory performed training as non-fulfillment, subject to owner/ADR confirmation.

The prior bootstrap-alignment gate is now closed by #455 commit `957f0265`.

## Validation / review state

- Child branch now contains exact current #455 head `957f0265499d2082536df64afc0682ad460ea7e7` via merge commit `04e36a81a5fdd3667d84b673fc16706673b11e4b`.
- Relative to #455, only the two intended implementation-analysis/plan documentation files differ.
- Branch comparison is **ahead 5 / behind 0** and GitHub reports #457 mergeable.
- No runtime code, policy version, baselines or tests are changed in this PR.
- Runtime validation commands and acceptance mapping are specified in the plan for the implementation stack.

## Dependency / merge intent

This PR should remain stacked on #455. The plan stays `Draft` while ADR-0038 is `Proposed`; runtime behavior remains blocked until #455 is accepted (or the owner explicitly authorizes implementation earlier).

### CODE REVIEW

File: docs/analysis/2026-09-07-adr-0038-recovery-placement-implementation-analysis.md [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+
+The required state is conceptually:
+
+```ts
+latestQualifyingRecoveryDate -> dueByDate = latest + 7 local calendar days
+```
+
+On `dueByDate`, urgency must already be tier 1 before ranking. When the latest date is unknown, a
+stable bootstrap epoch is needed; recomputing from `asOfDate` would let the deadline drift forever.
+
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/analysis/2026-09-07-adr-0038-recovery-placement-implementation-analysis.md]: Fixed. The docs now make `bootstrapDate` an explicit first deadline reference when no qualifying recovery is known: `dueByDate = bootstrapDate + 7` local calendar days. The bootstrap date is **not** recovery credit, the pre-bootstrap unknown prefix is excluded, and the first enforceable window is `B + 1 ... B + 7`; a real recovery before that deadline becomes the new reference. I also added the corresponding RP0/RP4 tests and an ADR-alignment gate because #455 currently says “stable bootstrap” without spelling out this boundary.

File: docs/analysis/2026-09-07-adr-0038-recovery-placement-implementation-analysis.md [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+**Recommendation:** introduce a small recovery-specific fact boundary, for example a
+`RecoveryPlacementFact` / `RecoveryDayOutcomeFact`, rather than widening performed-training truth.
+The exact name is implementation detail; the semantic separation is not.
+
+A useful source union would preserve identity rather than reducing everything to `completed`:
+
+```ts
+type RecoveryFactSource =
+  | { kind: 'performed_recovery'; performedOccurrenceId: string }
+  | {
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/analysis/2026-09-07-adr-0038-recovery-placement-implementation-analysis.md:279]: Fixed after tracing the current canonical-facts code. Exact coverage is keyed by canonical `workoutId`, while reverse workout→template inference is intentionally allowed to be ambiguous. `performed_recovery` now requires `performedOccurrenceId` + canonical `workoutId` + the recorded recovery qualification authority (`coverageSetId`/phase/`recovery_or_rest` or immutable equivalent); `templateId` is optional metadata only when authoritative/unambiguous. RP2 now requires replay to revalidate that stored identity instead of relying on an opaque occurrence-id lookup.

File: docs/plans/adr-0038-recovery-placement.md [hunk 3]
```diff
@@ -0,0 +1,708 @@
...
+A scenario result can answer **where the recovery contract was satisfied or missed and why**, not
+only how many Rest/Mobility days existed.
+
+---
+
+## RP5 — Knowledge governance, policy version and activation proof
+
+**Status:** planned  
+**Depends on:** RP1, RP3, RP4; must ship no later than live behavior activation  
+**Behavior change:** policy/replay metadata; final activation proof
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/plans/adr-0038-recovery-placement.md]: Fixed by separating implementation plumbing from activation. RP5 is now split into **RP5A** (knowledge registry/inventory/alignment, no live switch and no `POLICY_VERSION` bump) and **RP5B** (sole activation/replay-version boundary, depends on RP4 + RP5A). PR C stages ranking/D-SUPPORT + governance with production behavior default-off/disconnected; PR D first proves the exact rolling-window invariant and reviews corpus deltas, then connects the live path and bumps `POLICY_VERSION` in the same final boundary. Dependency graph, acceptance matrix, exit criteria and checklist were updated accordingly.

**Comment by Szczepanov** [comment]
Post-review update complete.

### What I changed

- **Bootstrap contract:** defined `bootstrapDate` as the initial deadline reference when recovery history is unknown: first due date is `B + 7`, `B` is not recovery credit, the unknown pre-bootstrap prefix is excluded, and the first enforceable window is `B + 1 ... B + 7`. Added corresponding pure/simulation test requirements and an explicit alignment gate with parent ADR #455.
- **Performed-recovery replay identity:** traced `performedTrainingFacts.ts` and made canonical `workoutId` + recovery qualification authority the required persisted/replay identity. `templateId` is optional only when direct/unambiguous because the current code intentionally refuses ambiguous workout→template inference.
- **ADR-0035 provenance:** clarified that `ExternalRestProvenance` is the canonical persisted/audit/replay contract; `ExternalRestDecisionProvenance` is its existing override-aware decision extension, not a second storage identity. The plan now names the replay/override adapter path explicitly.
- **Activation ordering:** split governance into **RP5A** (pre-activation registry/inventory/alignment; no live switch/version bump) and **RP5B** (sole activation/replay-version boundary). PR C now stages ranking/D-SUPPORT with production behavior off; PR D first proves the exact rolling invariant + reviews corpus deltas, then activates and bumps `POLICY_VERSION` in the same final review boundary.
- **PR description:** updated to reflect the revised contracts and safe A→B→C→D cut-line.

### Review/validation

- Replied to all 3 actionable CodeRabbit threads; CodeRabbit confirmed the fixes and all 3 threads are resolved.
- Also addressed the review-body provenance nitpick described above.
- GitHub reports this PR **mergeable**, 4 commits ahead / 0 behind its stacked #455 base, with only the two intended documentation files changed.
- Current commit status: **CodeRabbit success**.
- No runtime code, policy version, baselines, or app tests were changed/run because this PR is documentation-only.

### Evidence check

I re-checked the ADR’s recovery evidence boundary against the 2018 recovery consensus statement, the 2024 endurance-recovery umbrella review, and the 2025 Norwegian world-class-coach study. They support systematic/individualized load-recovery management, but not a universal one-recovery-per-seven-days physiological rule. The plan therefore correctly keeps the exact cadence as `product_policy` / heuristic rather than presenting it as physiological evidence.

### Remaining design gate (intentionally not invented here)

Before ADR-0038 activation, parent #455 should explicitly align with the new bootstrap boundary (`B + 7`, first enforceable window `B + 1 ... B + 7`) or select another explicit contract. The separate authored-Rest contradiction case (Rest directive present, no explicit override, but performed training later appears) also remains an owner/ADR decision before RP2 activation.

**Comment by Szczepanov** [comment]
Parent alignment gate is closed.

- #455 now defines the same bootstrap contract at `957f0265`: `B` is deadline-reference-only, `dueByDate = B + 7`, first evaluable window `B + 1 ... B + 7`, pre-bootstrap unknown prefix excluded.
- This branch was merged forward to that exact parent head in `04e36a81`.
- Current stack comparison: **ahead 5 / behind 0**; only the two intended analysis/plan docs differ from #455.
- PR metadata was refreshed to remove the now-stale “#455 still needs bootstrap alignment” gate.
- GitHub currently reports #457 mergeable; CodeRabbit status is green on the synchronized head.
